### PR TITLE
Read vendor-specific dead-key statuses as auth errors

### PR DESCRIPTION
**Pull request by dreikanter**
Changes:

- Move auth classification behind an `auth_error?` seam on `WebSearchProvider::Base`, so each provider can speak for its own vendor instead of sharing one status list
- Treat Brave's 422 `SUBSCRIPTION_TOKEN_INVALID` as a dead key, keyed on the error body since Brave reuses 422 for ordinary parameter complaints
- Treat Tavily's 432 (plan limit) and 433 (pay-as-you-go limit) as a spent key

### CODE REVIEW

File: app/services/web_search_provider/brave.rb [hunk 1]
```diff
@@ -2,9 +2,33 @@ module WebSearchProvider
...
+      return true if super
+      return false unless response.status == REJECTED_KEY_STATUS
+
+      rejected_key?(response.body)
+    end
+
+    # Matches the documented code, and any other failure Brave attributes to
+    # authentication, so a renamed token error still reads as a dead key.
+    def rejected_key?(body)
+      error = JSON.parse(body.to_s)["error"]
```
[the remark this answers is not in the document]
Comment (dreikanter) [reply app/services/web_search_provider/brave.rb]: Fixed in 9714530, before merge. `JSON.parse` does accept scalar documents (`JSON.parse("null")` → `nil`, `JSON.parse("[]")` → `[]`), so the concern was real: the body is now checked for a Hash before indexing, and a covering test exercises `null`, `[]` and `42` — all of which must come back as a plain `ProviderError`.


---
_Generated by [Claude Code](https://claude.ai/code)_
